api/METRICS.md

```diff
@@ -0,0 +1,114 @@
+<metric>_ratio(y_true, y_pred, *, sensitive_features, **other_kwargs)
+<metric>_group_min(y_true, y_pred, *, sensitive_features, **other_kwargs)
+<metric>_group_max(y_true, y_pred, *, sensitive_features, **other_kwargs)
+```
+
+*Transformations and transformation codes*
+
+|transformation function|output|metric-specific function|code|aif360|
+|-----------------------|------|------------------------|----|------|
+|`difference_from_summary`|max - min|`<metric>_difference`|D|unprivileged - privileged|
+|`ratio_from_summary`|min / max|`<metric>_ratio`|R| unprivileged / privileged|
+|`group_min_from_summary`|min|`<metric>_group_min`|Min| N/A |
+|`group_max_from_summary`|max|`<metric>_group_max`|Max| N/A |
+
+*Tasks and task codes*
+
+|task|definition|code|
+|----|----------|----|
+|binary classification|labels and predictions are in {0,1}|class|
+|probabilistic binary classification|labels are in {0,1}, predictions are in [0,1] and correspond to estimates of P(y\|x)|prob|
+|randomized binary classification|labels are in {0,1}, predictions are in [0,1] and represent the probability of drawing y=1 in a randomized decision|class-rand|
+|regression|labels and predictions are real-valued|reg|
+
+*Predefined metric-specific functions*
+
+* variants: D, R, Min, Max refer to the transformations from the table above; G refers to `<metric>_group_summary`.
+
+|metric|variants|task|notes|aif360|
+|------|--------|-----|----|------|
+|`selection_rate`| G,D,R,Min | class | | &#x2713; |
+|`demographic_parity`| D,R | class | `selection_rate_difference`, `selection_rate_ratio` | `statistical_parity_difference`, `disparate_impact`|
+|`accuracy_score`| G,D,R,Min | class | sklearn | `accuracy` |
+|`balanced_accuracy_score` | G | class | sklearn | - |
+|`mean_absolute_error` | G,D,R,Max | class, reg | sklearn | class only: `error_rate` |
+|`confusion_matrix` | G | class | sklearn | `binary_confusion_matrix` |
+|`false_positive_rate` | G,D,R | class | | &#x2713; |
+|`false_negative_rate` | G | class | | &#x2713; |
+|`true_positive_rate` | G,D,R | class | | &#x2713; |
+|`true_negative_rate` | G | class | | &#x2713; |
+|`equalized_odds` | D,R | class | max of difference or ratio under `true_positive_rate`, `false_positive_rate` | - |
+|`precision_score`| G | class | sklearn | &#x2713; |
+|`recall_score`| G | class | sklearn | &#x2713; |
+|`f1_score`| G | class | sklearn | - |
+|`roc_auc_score`| G | prob | sklearn | - |
+|`log_loss`| G | prob | sklearn | - |
+|`mean_squared_error`| G | prob, reg | sklearn | - |
+|`r2_score`| G | reg | sklearn | - |
+
+## Dashboard questions
+
+1. Should we enable regression metrics for probabilistic classification?
+   * `mean_absolute_error`, `mean_squared_error`, `mean_squared_error(...,squared=False)`
+1. Should we introduce balanced error metrics for probabilistic classification?
+   * `balanced_mean_{squared,absolute}_error`, `balanced_log_loss`
+1. Do we keep `mean_prediction` and `mean_{over,under}prediction`?

```

Comment:
Unless we no longer need it in the dashboard, it will need to be available in python. We could special case these and not have them callable by the user if we want. (what about selection rate? the other ones like our differently normalized underprediction rate)?

Reply:
For the time being, I'll proceed with non-breaking changes. We will need to review metrics / dashboard integration in a separate PR.